### PR TITLE
chore(NotificationDrawer): demo conversion from JS to TS

### LINKED ISSUE #9438: Convert notification drawer react demos to TS

### PR DESCRIPTION
**Pull request by adamviktora**
**What**: Closes #9438

### CODE REVIEW

**Comment by nicolethoen** [comment]
I cannot figure out why the ts -> js button is not appearing in the second demo and the site doesn't seem to think the second demo is written in ts... still looking into it, but maybe someone else here knows?

File: packages/react-core/src/demos/NotificationDrawer/examples/NotificationDrawerGrouped.tsx [hunk 1]
```diff
@@ -0,0 +1,796 @@
...
+      <DropdownItem key="settings">Settings</DropdownItem>
+    </>
+  );
+  const notificationDrawerDropdownItems = (
+    <>
+      <DropdownItem
+        key="link"
+        to="#"
+        // Prevent default onClick behavior for demo purposes
+        onClick={(ev: { preventDefault: () => any }) => ev.preventDefault()}
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/NotificationDrawer/examples/NotificationDrawerGrouped.tsx:372]: This is causing a Syntax error when opening the demo. Can we type the `ev` as any to match the `onClick` prop in DropdownItem?

```suggestion
        onClick={(ev: any) => ev.preventDefault()}
```

File: packages/react-core/src/demos/NotificationDrawer/examples/NotificationDrawerBasic.tsx [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+import CogIcon from '@patternfly/react-icons/dist/esm/icons/cog-icon';
+import HelpIcon from '@patternfly/react-icons/dist/esm/icons/help-icon';
+import QuestionCircleIcon from '@patternfly/react-icons/dist/esm/icons/question-circle-icon';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import BarsIcon from '@patternfly/react-icons/dist/js/icons/bars-icon';
+import imgAvatar from '@patternfly/react-core/src/components/Avatar/examples/avatarImg.svg';
+import SearchIcon from '@patternfly/react-icons/dist/esm/icons/search-icon';
+import pfLogo from '@patternfly/react-core/src/demos/assets/pf-logo.svg';
+
+export const BasicNotificationDrawer: React.FunctionComponent = () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/NotificationDrawer/examples/NotificationDrawerBasic.tsx:61]: Can we just update these exports to match the file names, similar to our examples? `NotificationDrawerBasic` and `NotificationDrawerGrouped`

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Haven't looked into the above issue that Nicole mentioned, but I'd imagine we could update the `js file=...` to `ts file=...` in the MD file. Do you get the same issue as the Nav demo PR when doing so?

**Comment by thatblindgeye** [comment]
@nicolethoen I wonder if it's just an issue with the preview... I tried editing my Org clone locally to add in the tsx example files to react-core (using `js file=...` in the MD), and the demos rendered with the correct "JS" icon in the example toolbar.

**Comment by kmcfaul** [comment]
Maybe a rebase/rebuild would iron out the surge?

**Review by nicolethoen — APPROVED** [review]
looks like its all working now!

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**